src/backend/replication/libpqwalreceiver/libpqwalreceiver.c

```diff
@@ -124,10 +124,9 @@ libpqrcv_connect(char *conninfo)
 		int			io_flag;
 		int			rc;
 
-		if (status == PGRES_POLLING_READING)
-			io_flag = WL_SOCKET_READABLE;
-		else
-			io_flag = WL_SOCKET_WRITEABLE;
+		io_flag = WL_SOCKET_READABLE;
```

Comment:
Add comment that this bit have to be setup for successive call of  `WaitLatchOrSocket()`